src/main/java/frc/team997/lib/trajectory/AsymmetricTrapezoidProfile.java

```diff
@@ -279,83 +222,11 @@ public AsymmetricTrapezoidProfile(Constraints constraints, State target, State i
-     * @param time The time since the beginning of the profile.
-     * @return The position and velocity of the profile at the time.
-     */
-    public State calculate(double time) {
-        double position = initial.position;
-        for (ProfilePhase phase : phases) {
-            if (time - phase.time < 0) {
-                return new State(
-                        position
-                                + 0.5 * phase.acceleration * Math.pow(time, 2)
-                                + phase.initialVelocity * time,
-                        time * phase.acceleration + phase.initialVelocity);
-            } else {
-                time -= phase.time;
-                position += phase.position;
-            }
-        }
-        return new State(position, target.velocity);
-    }
-    /**
-     * Returns the time as a function of the target position.
-     *
-     * @param targetPosition The target position.
-     * @return The time left until a target position in the profile is reached. Returns NaN if
-     *     targetPosition < initial.position, and the total time if targetPosition > target.position
-     */
-    public double timeLeftUntil(double targetPosition) {
-        targetPosition = targetPosition - initial.position;
-        double time = 0;
-        for (ProfilePhase phase : phases) {
-            if ((targetPosition - phase.position) * direction < 0) {
-                if (phase.acceleration == 0) {
-                    time += targetPosition / constraints.maxVelocity;
-                } else {
-                    double sqrt =
-                            Math.sqrt(
-                                    Math.pow(phase.initialVelocity, 2)
-                                            + 2 * phase.acceleration * targetPosition);
-                    time += (-phase.initialVelocity + direction * sqrt) / phase.acceleration;
-                }
-                return time;
-            } else {
-                time += phase.time;
-                targetPosition -= phase.position;
-            }
-        }
-        return time;
-    }
-    /**
-     * Returns the total time the profile takes to reach the goal.
-     *
-     * @return The total time the profile takes to reach the goal.
-     */
-    public double totalTime() {
-        double time = 0;
-        for (ProfilePhase phase : phases) {
-            time += phase.time;
-        }
-        return time;
-    }
-    /**
-     * Returns true if the profile has reached the goal.
-     *
-     * <p>The profile has reached the goal if the time since the profile started has exceeded the
-     * profile's total time.
-     *
-     * @param time The time since the beginning of the profile.
-     * @return True if the profile has reached the goal.
-     */
-    public boolean isFinished(double time) {
-        return time >= totalTime();
+        ArrayList<ProfilePhase> validPhases = new ArrayList<ProfilePhase>();
+        if (accelPhase.time > 0) validPhases.add(accelPhase);
```

Comment:
no need to create a new ArrayList here - calling super.phases.add would be better. Then, we can make MotionProfile.phases final (make sure to assign it to an empty ArrayList in the MotionProfile class. Also, I'd like to when possible move away from separating variable declaration from initialization (an initialization on a non-static member field will initialize at construction anyways so the only reason initialization should go in the constructor is if there has to be any processing before the construction IMO)